components/dashboard/src/settings/PersonalAccessTokens.tsx

```diff
@@ -61,26 +154,45 @@ export function PersonalAccessTokenCreateView() {
         setValue({ ...value, ...change });
     };
 
-    const createToken = async () => {
+    const regenerate = async () => {
+        if (!editTokenID) {
+            return;
+        }
+        try {
+            const resp = await personalAccessTokensService.regeneratePersonalAccessToken({
+                id: editTokenID,
+                expirationTime: Timestamp.fromDate(value.expirationDate),
+            });
+            backToListView({ method: Method.Regerenrate, data: resp.token! });
+        } catch (e) {
+            setErrorMsg(e.message);
+        }
+    };
+
+    const handleConfirm = async () => {
         if (value.name.length < 3) {
             setErrorMsg("Token Name should have at least three characters.");
             return;
         }
         try {
-            const resp = await personalAccessTokensService.createPersonalAccessToken({
-                token: {
-                    name: value.name,
-                    expirationTime: Timestamp.fromDate(value.expirationDate),
-                    scopes: ["function:*", "resource:default"],
-                },
-            });
-            history.push({
-                pathname: settingsPathPersonalAccessTokens,
-                state: {
-                    method: "CREATED",
-                    data: resp.token,
-                },
-            });
+            const resp = editTokenID
+                ? await personalAccessTokensService.updatePersonalAccessToken({
+                      token: {
+                          id: editTokenID,
+                          name: value.name,
+                          scopes: ["function:*", "resource:default"],
+                      },
+                      updateMask: { paths: ["name", "scopes"] },
```

Comment:
Does this need to determine what's changed, to properly apply the update mask? Or do we just update all? There is a potential risk to two simultaneous requests performing an update but given these are Personal Tokens, we can probably keep it this way.